superset/datasets/schemas.py

```diff
@@ -42,26 +43,25 @@
 }
 
 
-def validate_python_date_format(value: str) -> None:
-    regex = re.compile(
-        r"""
-        ^(
-            epoch_s|epoch_ms|
-            (?P<date>%Y([-/]%m([-/]%d)?)?)([\sT](?P<time>%H(:%M(:%S(\.%f)?)?)?))?
-        )$
-        """,
-        re.VERBOSE,
-    )
-    match = regex.match(value or "")
-    if not match:
-        raise ValidationError([_("Invalid date/timestamp format")])
+def validate_python_date_format(value: str) -> bool:
+    if value in ("epoch_s", "epoch_ms", None):
+        return True
+    try:
+        dt_str = datetime.now().strftime(value)
+        isoparse(dt_str)
```

Comment:
```suggestion
        isoparse(value)
```

Reply:
The `isoparse(value)` is not working correctly. 
The `value` passed to it is a string pattern, while the `isoparse` function is intended to parse datetime strings into datetime objects.
Therefore, I am using `strftime` to generate a datetime string based on the provided string pattern.
Subsequently, I use `isoparse` to parse the string into a datetime object for validating ISO 8601.

Reply:
Thanks @mapledan for the explanation. I forgot that `value` was a `strptime` format as opposed to a serialized datetime.